source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -413,6 +419,43 @@ def event_UIA_notification(self, activityId=None, **kwargs):
 			return
 		super(WordDocument, self).event_UIA_notification(**kwargs)
 
+	# The following overide of the EditableText._caretMoveBySentenceHelper private method
+	# Falls back to the MS Word object model if available.
+	# This override should be removed as soon as UI Automation in MS Word has the ability to move by sentence.
+	def _caretMoveBySentenceHelper(self, gesture, direction):
+		if isScriptWaiting():
+			return
+		if not self.WinwordSelectionObject:
+			# Legacy object model not available.
+			gesture.send()
+			return
+		# Using the legacy object model,
+		# Move the caret to the next sentence in the requested direction.
+		legacyInfo = LegacyWordDocumentTextInfo(self, textInfos.POSITION_CARET)
+		legacyInfo.move(textInfos.UNIT_SENTENCE, direction)
+		legacyInfo.updateCaret()

```

Comment:
I think we can avoid one caret update by removing this one and create the UIA info for the end of the sentence before creating the info for the start, however, that means all caret updates will be performed by the object model. I'm not sure how costly caret updates are.

Reply:
I don't quite follow. You say we can remove a caret update, yet then you say but you are not sure if object model caret updates are slow... are you proposing to add another caret update via the ojbect model?
I'd want to limit the amount of object model calls most definitely.  
If you could provide a code example that might help.

Reply:
Could you have a look at http://github.com/leonardder/nvda/commit/40e2fb8c32d068a3a36f3e86f99f8d3d2d1e79db ?

Reply:
Yep, that makes sense. I'll add that commit.